File: repos_estudiantes_ALVAREZ_tercera_entrega/HUANCA_CARDOZO_ADRIAN_GUSTAVO/PERSISTENCIA_DE_OBJETOS/EJERCICIO_5/Farmacia.java
package PERSISTENCIA_DE_OBJETOS.EJERCICIO_5;
import java.io.Serializable;
import java.util.ArrayList;

public class Farmacia implements Serializable {
    private String nombreFarmacia;
    private int sucursal;
    private String direccion;
    private int nroMedicamentos;
    private ArrayList<Medicamento> medicamentos;

    public Farmacia() {
        medicamentos = new ArrayList<>();
    }

    public void leer(String nombreFarmacia, int sucursal, String direccion) {
        this.nombreFarmacia = nombreFarmacia;
        this.sucursal = sucursal;
        this.direccion = direccion;
    }

    public void mostrar() {
        System.out.println("Farmacia: " + nombreFarmacia + ", Sucursal: " + sucursal + ", Direccion: " + direccion);
        mostrarMedicamentos();
    }

    public String getDireccion() {
        return direccion;
    }

    public int getSucursal() {
        return sucursal;
    }

    public void agregarMedicamento(Medicamento m) {
        medicamentos.add(m);
        nroMedicamentos++;
    }

    public void mostrarMedicamentos() {
        for (Medicamento m : medicamentos) {
            m.mostrar();
        }
    }

    public void mostrarMedicamentos(String tipoFiltro) {
        for (Medicamento m : medicamentos) {
            if (m.getTipo().equalsIgnoreCase(tipoFiltro)) {
                m.mostrar();
            }
        }
    }

    public boolean buscaMedicamento(String nombreMedicamento) {
        for (Medicamento m : medicamentos) {
            if (m.getNombre().equalsIgnoreCase(nombreMedicamento)) {
                return true;
            }
        }
        return false;
    }
}

File: repos_estudiantes_TARQUI_tercera_entrega/RUBIN_DE_CELIS_VALDEZ_ANDRES/PERSISTENCIA/EJERCICIO-5/Farmacia.py
from Medicamento import Medicamento

class Farmacia:
    def __init__(self):
        self.nombreFarmacia = ""
        self.sucursal = 0
        self.direccion = ""
        self.numMedicamentos = 0
        self.medicamentos = []
    
    def leer(self):
        self.nombreFarmacia = input("Nombre de la farmacia: ")
        self.sucursal = int(input("Número de sucursal: "))
        self.direccion = input("Dirección: ")
        self.numMedicamentos = int(input("Número de medicamentos: "))
        
        for _ in range(self.numMedicamentos):
            med = Medicamento()
            med.leer()
            self.medicamentos.append(med)
    
    def mostrar(self):
        print(f"\nFarmacia: {self.nombreFarmacia}")
        print(f"Sucursal: {self.sucursal}")
        print(f"Dirección: {self.direccion}")
        print("\nMedicamentos disponibles:")
        for med in self.medicamentos:
            med.mostrar()
    
    def getDireccion(self):
        return self.direccion
    
    def getSucursal(self):
        return self.sucursal
    
    def mostrarMedicamentos(self, tipo):
        print(f"\nMedicamentos para {tipo}:")
        for med in self.medicamentos:
            if med.getTipo().lower() == tipo.lower():
                med.mostrar()
    
    def buscarMedicamento(self, nombre):
        for med in self.medicamentos:
            if med.nombre.lower() == nombre.lower():
                return med
        return None
    
    def to_dict(self):
        """Convierte el objeto Farmacia a un diccionario"""
        return {
            'nombreFarmacia': self.nombreFarmacia,
            'sucursal': self.sucursal,
            'direccion': self.direccion,
            'numMedicamentos': self.numMedicamentos,
            'medicamentos': [med.to_dict() for med in self.medicamentos]
        }
    
    @classmethod
    def from_dict(cls, data):
        farmacia = cls()
        farmacia.nombreFarmacia = data['nombreFarmacia']
        farmacia.sucursal = data['sucursal']
        farmacia.direccion = data['direccion']
        farmacia.numMedicamentos = data['numMedicamentos']
        farmacia.medicamentos = [Medicamento.from_dict(med) for med in data['medicamentos']]
        return farmacia